src/AppInstallerCLICore/Workflows/PinFlow.cpp

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "Resources.h"
+#include "PinFlow.h"
+#include "TableOutput.h"
+#include "Microsoft/PinningIndex.h"
+#include "Microsoft/SQLiteStorageBase.h"
+#include "winget/RepositorySearch.h"
+
+using namespace AppInstaller::Repository;
+using namespace AppInstaller::Repository::Microsoft;
+
+namespace AppInstaller::CLI::Workflow
+{
+    namespace
+    {
+        // Creates a Pin appropriate for the context based on the arguments provided
+        Pinning::Pin CreatePin(Execution::Context& context, std::string_view packageId, std::string_view sourceId, const std::string& installedVersion)
+        {
+            if (context.Args.Contains(Execution::Args::Type::GatedVersion))
+            {
+                return Pinning::Pin::CreateGatingPin({ packageId, sourceId }, context.Args.GetArg(Execution::Args::Type::GatedVersion));
+            }
+            else if (context.Args.Contains(Execution::Args::Type::BlockingPin))
+            {
+                return Pinning::Pin::CreateBlockingPin({ packageId, sourceId }, { installedVersion });
+            }
+            else
+            {
+                return Pinning::Pin::CreatePinningPin({ packageId, sourceId }, { installedVersion });
+            }
+        }
+    }
+
+    void OpenPinningIndex(Execution::Context& context)
+    {
+        auto pinningIndex = PinningIndex::OpenOrCreateDefault(SQLiteStorageBase::OpenDisposition::ReadWrite);
+        context.Add<Execution::Data::PinningIndex>(std::make_shared<PinningIndex>(std::move(pinningIndex)));
+    }
+
+    void GetAllPins(Execution::Context& context)
+    {
+        AICLI_LOG(CLI, Info, << "Getting all existing pins");
+        context.Add<Execution::Data::Pins>(context.Get<Execution::Data::PinningIndex>()->GetAllPins());
+    }
+
+    void SearchPin(Execution::Context& context)

```

Comment:
I think I'll remove the comments from the other workflow tasks instead. From what I see, we generally keep the comments in the headers. The others have comments only because I copied the definitions from the header and I forgot to remove them.

Reply:
Works for me!